src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx

```diff
@@ -2699,4 +2741,97 @@ void HwEmShim::closemMessengerThread() {
 }
 /********************************************** QDMA APIs IMPLEMENTATION END**********************************************/
 /**********************************************HAL2 API's END HERE **********************************************/
+
+/********************************************** Q2H_helper class implementation starts **********************************************/
+bool device2xrt_trans_cb(bool, int, unsigned long int, void*,unsigned long int) {
+    std::cout << " device2xrt_trans_cb() is called...." <<std::endl;
+    return true;
+}
+Q2H_helper :: Q2H_helper() : buf_size(0),buf(NULL), r(0) {
+    std::cout << " Start of Q2H_helper()..." << std::endl;
+    ci_msg = new call_packet_info;
+    ri_msg = new response_packet_info;
+    Q2h_sock = new unix_socket("xcl_sock_K2H");
```

Comment:
Socket is not getting freed

Reply:
Now Deleting it in destructor of  Q2H_helper class.

Reply:
Which is exactly why one should not use raw calls to new.

Reply:
Now smart pointer are in use which replaced "new"